yaml-validator/src/types/object.rs

```diff
@@ -336,3 +376,58 @@ mod tests {
         );
     }
 }
+
+#[cfg(test)]
+mod test_required_field {
+    use super::*;
+    use crate::utils::load_simple;
+    use crate::SchemaObject;
+
+    #[cfg(feature = "smallvec")]
+    use smallvec::smallvec;
+
+    #[test]
+    fn from_yaml() {
+        SchemaObject::try_from(&load_simple(
+            r#"
+            items:
+              something:
+                type: string
+            required: []
+        "#,
+        ))
+        .unwrap();
+    }
+
+macro_rules! add_creation_fail_tests {
```

Comment:
Are you accept macro in tests? Or expand it?

Reply:
I think for very repetitive and boring tests like these, that someone likely won't use for documentation about behaviour of the code, but are just there to prevent future regressions, macros are absolutely fine, and should be implemented more broadly in the whole codebase.

I think making some macros for testing the wrong type error for example would make a lot of sense, but for now, I think this use case is great.